TombEngine/Specific/EngineMain.cpp

```
@@ -0,0 +1,521 @@
+			continue;
+
+		if (g_GameScript == nullptr)
+		{
+			TENLog("Scripting engine not initialized.", LogLevel::Error);
+			continue;
+		}
+		else
+		{
+			g_GameScript->AddConsoleInput(input);
+		}
+	}
+
+	return 0;
+}
+
+static void HandleWindowFocusGained(SDL_Window* window)
+{
+	SetInputLockState(false);
+
+	if (!g_Configuration.EnableWindowedMode)
+	{
+		g_Renderer.ToggleFullScreen(true);
+	}
+
+	if (ThreadSuspendCount > 0)
+	{
+		TENLog("Resuming game thread", LogLevel::Info);
+
+		if (!g_VideoPlayer.Resume())
+			ResumeAllSounds(SoundPauseMode::Global);
+
+		ResumeGameThread();
+	}
+}
+
+static void HandleWindowFocusLost(SDL_Window* window)
+{
+	SetInputLockState(true);
+
+	if (!g_Configuration.EnableWindowedMode)
+		SDL_MinimizeWindow(window);
+
+	bool isMinimized =
+		(SDL_GetWindowFlags(window) & SDL_WINDOW_MINIMIZED) != 0;
+
+	if ((!DebugMode || isMinimized) && ThreadSuspendCount == 0)
+	{
+		TENLog("Suspending game thread", LogLevel::Info);
+
+		if (!g_VideoPlayer.Pause())
+			PauseAllSounds(SoundPauseMode::Global);
+
+		PauseGameThread();
+	}
+}
+
+void PauseGameThread()
+{
+	SDL_LockMutex(GamePauseMutex);
+
+	ThreadSuspendCount++;
+	GamePaused = true;
+
+	SDL_UnlockMutex(GamePauseMutex);
+}
+
+void ResumeGameThread()
+{
+	SDL_LockMutex(GamePauseMutex);
+
+	if (ThreadSuspendCount > 0)
+		ThreadSuspendCount--;
+
+	if (ThreadSuspendCount == 0)
+	{
+		GamePaused = false;
+		SDL_BroadcastCondition(GamePauseCond);
+	}
+
+	SDL_UnlockMutex(GamePauseMutex);
+}
+
+void WaitIfGamePaused()
+{
+	SDL_LockMutex(GamePauseMutex);
+	while (GamePaused && !ThreadEnded && DoTheGame)
+	{
+		SDL_WaitCondition(GamePauseCond, GamePauseMutex);
+	}
+	SDL_UnlockMutex(GamePauseMutex);
+}
+
+int main(int argc, char* argv[])
+{
+	g_Platform = CreatePlatformSubsystem();
+	g_Platform->Initialize();
+	g_Platform->CheckPrerequisites();
+	
+	// Initialize SDL3
+	if (!SDL_Init(SDL_INIT_VIDEO | SDL_INIT_AUDIO | SDL_INIT_EVENTS))
+	{
+		// handle error
```

I guess we need to at least report an error here via non-TENLog means?